src/util/asset_management.py

```diff
@@ -2,34 +2,69 @@
 
 import os
 from pathlib import Path
-from typing import Callable
+from typing import Callable, Union
 
+from src.config.lockfile_entry import LockfileEntry
 from src.lib.multikey_dict import MultiKeyDict
 from src.util.web import download_file
 
 
-def install_assets(
-        asset_entries: MultiKeyDict, install_queue: set, asset_path: Path,
+def create_entry_queue(
+        lockfile_entries: MultiKeyDict, to_install: set) -> list[LockfileEntry]:
+    """Given a lookup dictionary of lockfile entries and a set of asset
+    references, return a list of the associated LockfileEntry objects."""
+    queue = []
+    for queued_asset in to_install:
+        queue.append(lockfile_entries.get_by_multikey(queued_asset))
+    return queue
+
+
+def install_asset(
+        lockfile_entry: LockfileEntry, asset_path: Path,
         echo: Callable):
+    """Installs the given asset to the asset path."""
+    cdn_link = lockfile_entry.asset.cdn_link
+    file_path = asset_path / lockfile_entry.name
+    download_file(cdn_link, file_path)
```

Comment:
Maybe the hash checking should be done in download_file against the buffer before it is written to disk. It seems inefficient to write to disk and uninstall if the hash mismatches